src/H5FDmpio.c

```diff
@@ -1865,7 +1865,7 @@ H5FD__mpio_vector_build_types(uint32_t count, H5FD_mem_t types[], haddr_t addrs[
                 if (!sub_types) {
                     HDassert(!sub_types_created);
 
-                    if (NULL == (sub_types = (int *)HDmalloc((size_t)count * sizeof(MPI_Datatype))))
+                    if (NULL == (sub_types = HDmalloc((size_t)count * sizeof(MPI_Datatype))))
```

Comment:
sub_types is an MPI_Datatype *, not an int *

Reply:
There's no need to cast the allocated pointers in C anyways.